pkg/build/util/generate.go

```diff
@@ -45,6 +49,69 @@ func GenerateBuildFromBuild(build *buildapi.Build) *buildapi.Build {
 	}
 }
 
+// GenerateBuildWithImageTag generates a build definition based on the current imageid
```

Comment:
imageid if it's our docker registry w/ the immutable imageid tags, otherwise it's a bit less clear, but "current image" isn't really accurate, that's what we already had even w/o this functionality.